File: LAB_CN/LAB_CN/Python/subnet.py
def main():
    ip = input("Enter IP Address: ")
    classip = determine_class(ip)
    print("Class:", classip)
    subnets = int(input("Enter the number of subnets: "))
    subnet_mask(ip, subnets)

def determine_class(ip):
    first_octet = int(ip.split(".")[0])
    if 1 <= first_octet <= 126:
        return 'A'
    elif 128 <= first_octet <= 191:
        return 'B'
    elif 192 <= first_octet <= 223:
        return 'C'
    elif 224 <= first_octet <= 239:
        return 'D'
    else:
        return 'E'

def subnet_mask(ip, subnets):
    octets = list(map(int, ip.split(".")))
    
    subnet_size = 256 // subnets
    print("Subnet Ranges:")
    for i in range(subnets):
        start = i * subnet_size
        end = (i + 1) * subnet_size - 1
        print("Subnet %d: %d.%d.%d.%d - %d.%d.%d.%d" % (
            i + 1, octets[0], octets[1], octets[2], start, octets[0], octets[1], octets[2], end
        ))

if __name__ == "__main__":
    main()

File: LAB_CN/LAB_CN/Java/Subnetting/Subnet.java
import java.util.Scanner;

public class Subnet {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter IP Address: ");
        String ip = scanner.nextLine();
        char classip = determineClass(ip);
        System.out.println("Class: " + classip);
        System.out.print("Enter the number of subnets: ");
        int subnets = scanner.nextInt();
        subnetMask(ip, subnets);
        scanner.close();
    }

    public static char determineClass(String ip) {
        String[] octets = ip.split("\\.");
        int first_octet = Integer.parseInt(octets[0]);

        if (first_octet >= 1 && first_octet <= 126)
            return 'A';
        else if (first_octet >= 128 && first_octet <= 191)
            return 'B';
        else if (first_octet >= 192 && first_octet <= 223)
            return 'C';
        else if (first_octet >= 224 && first_octet <= 239)
            return 'D';
        else
            return 'E';
    }

    public static void subnetMask(String ip, int subnets) {
        String[] octets = ip.split("\\.");
        int[] octet = new int[4];

        for (int i = 0; i < 4; i++) {
            octet[i] = Integer.parseInt(octets[i]);
        }

        int bits = 0;
        while ((1 << bits) < subnets)
            bits++;

        int[] mask = new int[4];
        for (int i = 0; i < 4; i++) {
            if (bits >= 8) {
                mask[i] = 255;
                bits -= 8;
            } else if (bits > 0) {
                mask[i] = (255 << (8 - bits)) & 255;
                bits = 0;
            } else {
                mask[i] = 0;
            }
        }

        System.out.println("Subnet Ranges:");
        for (int i = 0; i < subnets; i++) {
            int start = i * (256 / subnets);
            int end = (i + 1) * (256 / subnets) - 1;
            System.out.printf("Subnet %d: %d.%d.%d.%d - %d.%d.%d.%d%n", i + 1,
                    octet[0], octet[1], octet[2], start,
                    octet[0], octet[1], octet[2], end);
        }
    }
}